keepmenu

```diff
@@ -63,16 +63,19 @@ def gen_passwd(length=20, use_digits=True, use_spec_chars=True):
     Returns: password - string
 
     """
-    length = length if length >= 4 else 4
-    dig = string.digits if use_digits is True else ''
-    spec = string.punctuation if use_spec_chars is True else ''
-    alphabet = string.ascii_letters + dig + spec
+    default_length = len(alphabet_list) if len(alphabet_list) > 4 else 4
+    length = length if length >= default_length else default_length
```

Comment:
I think this should just be `length = max(4, length)` to enforce a minimum length of 4. The problem with setting a minimum length as `len(alphabet_list)` would happen with the default settings, which would enforce a min password length of 8. If someone is trying to generate a password of length 6 for an old sketchy system (they still exist :( ) then it should allow a length of 6. However, we have to figure out the best way of notifying the user that not all character sets will be used. Perhaps instead of returning `None` on failure we could return a particular error like `Error: not all character sets used. Please edit config.ini`, and then at line 1152 use `if sel.startswith("Error:")` instead of `if not sel` to display the particular error returned.

Reply:
Understood, please disregard my latest change ;)